code/atom.dm

```diff
@@ -632,10 +632,16 @@
 /atom/proc/attack_ai(mob/user as mob)
 	return
 
-//mbc : sorry, i added a 'is_special' arg to this proc to avoid race conditions.
-/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
+///wrapper proc for /atom/proc/attackby so that signals are always sent. Call this, but do not override it.
+/atom/proc/Attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
+	SHOULD_NOT_OVERRIDE(1)
 	if(SEND_SIGNAL(src,COMSIG_ATTACKBY,W,user))
 		return
+	src.attackby(W, user, params, is_special)
+
+//mbc : sorry, i added a 'is_special' arg to this proc to avoid race conditions.
+///internal proc for when an atom is attacked by an item. Override this, but do not call it,
+/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
 	src.material?.triggerOnHit(src, W, user, 1)
```

Comment:
```suggestion
/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
	PROTECTED_PROC(TRUE)
	src.material?.triggerOnHit(src, W, user, 1)
```
possibly? won't prevent people from calling it from within the type but will provide some level of protection at the very least